<?xml version="1.0"?>
<launch>
 
<param name="robot_description_board" command="$(find xacro)/xacro '$(find franka_cal_sim)/xacro/aprilgrid.xacro'" />

<include file="$(find franka_cal_sim)/launch/franka.launch">
</include>

<node pkg="nodelet" type="nodelet" name="standalone_nodelet"  args="manager" output="screen"/>



<node name="image_view" pkg="image_view" type="image_view" args="image:=/simple_camera/image_raw"/>

<node name="estimator" pkg="franka_cal_sim" type="estimation_service_server_new.py" output="screen"/>

<node name="rl_client" pkg="franka_cal_sim" type="RL_algo_long_test.py" output="screen">
<rosparam command="load" file="$(find franka_cal_sim)/config/config.yaml" />
</node>

<node pkg="nodelet" type="nodelet" name="simulation_server" args="load franka_cal_sim/simulation_server standalone_nodelet" output="screen">
</node>


<node pkg="nodelet" type="nodelet" name="model_client" args="load franka_cal_sim/model_client standalone_nodelet" output="screen">
</node>

<!-- Spawn the checkbaord into Gazebo -->
<node name="spawn_sdf" pkg="gazebo_ros" type="spawn_model" args="-file $(find franka_cal_sim)/models/small_checkerboard/small_checkerboard.sdf -sdf -model small_checkerboard -x -0.75 -y -0.35 -z 0.5 -P 1.57 -unpause" />

</launch>
